Server/Server.Core/Server.Core.Static/ArgParser.hpp

```diff
@@ -67,8 +67,8 @@ class ArgParser
             .default_value<std::string>("juniorgram")
             .help("set the name of database")
             .action([](const std::string& value) {
-                const std::vector<std::string> choices = {"juniorgram", "testdb"};
-                if (std::find(choices.begin(), choices.end(), value) != choices.end())
+                if (const std::vector<std::string> choices = {"juniorgram", "testdb"};
+                    std::find(choices.begin(), choices.end(), value) != choices.end())
```

Comment:
What is the reason for this decision?
Is the declaration and implementation in the conditional part a good practice?

Reply:
C++17 introduced a construct to create and initialize a variable within the condition of if and switch statements and C++20 added this construct to range-based for loops. Using this new feature simplifies common code patterns and helps in giving variables the right scope.